koku/masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -103,30 +111,35 @@ def create_daily_archives(
+    use_cols, time_interval, process_date = get_processing_date(
         local_file, s3_csv_path, manifest_id, provider_uuid, start_date, end_date, context, tracing_id
     )
-    intervals = data_frame[time_interval].unique()
-    for interval in intervals:
-        date = interval.split("T")[0]
-        csv_date = datetime.datetime.strptime(date, "%Y-%m-%d")
-        # Adding end here so we dont bother to process future incomplete days (saving plan data)
-        if csv_date >= process_date and csv_date <= end_date:
-            dates.add(date)
-    if not dates:
-        return [], {}
-    directory = os.path.dirname(local_file)
-    date_range = {"start": min(dates), "end": max(dates), "invoice_month": None}
-    data_frame = data_frame[data_frame[time_interval].str.contains("|".join(dates))]
-    for date in dates:
-        daily_data = data_frame[data_frame[time_interval].str.match(date)]
-        day_file = ReportManifestDBAccessor().update_and_get_day_file(date, manifest_id)
-        day_filepath = f"{directory}/{day_file}"
-        daily_data.to_csv(day_filepath, index=False, header=True)
-        utils.copy_local_report_file_to_s3_bucket(
-            tracing_id, s3_csv_path, day_filepath, day_file, manifest_id, start_date, context
-        )
-        daily_file_names.append(day_filepath)
+    LOG.info(log_json(tracing_id, msg="pandas read csv with following usecols", usecols=use_cols, context=context))
+    with pd.read_csv(local_file, usecols=use_cols, chunksize=settings.PARQUET_PROCESSING_BATCH_SIZE) as reader:
+        for i, data_frame in enumerate(reader):
+            if data_frame.empty:
+                continue
+            intervals = data_frame[time_interval].unique()
+            for interval in intervals:
+                date = interval.split("T")[0]
+                csv_date = datetime.datetime.strptime(date, "%Y-%m-%d")
+                # Adding end here so we dont bother to process future incomplete days (saving plan data)
+                if csv_date >= process_date and csv_date <= end_date:
+                    dates.add(date)
+            if not dates:
+                return [], {}
+            directory = os.path.dirname(local_file)
+            date_range = {"start": min(dates), "end": max(dates), "invoice_month": None}
+            data_frame = data_frame[data_frame[time_interval].str.contains("|".join(dates))]
+            for date in dates:
+                daily_data = data_frame[data_frame[time_interval].str.match(date)]
+                day_file = ReportManifestDBAccessor().update_and_get_day_file(date, manifest_id)
+                day_filepath = f"{directory}/{day_file}_{i}.csv"
+                daily_data.to_csv(day_filepath, index=False, header=True)
+                utils.copy_local_report_file_to_s3_bucket(
+                    tracing_id, s3_csv_path, day_filepath, day_file, manifest_id, start_date, context
+                )
+                daily_file_names.append(day_filepath)
```

Comment:
Here and in the Azure method, we do a LOT of stuff with dates, and it's difficult to follow. When we read CSV files, we can pass the `parse_dates` arg which is a list of columns containing dates. Then we have an actual date-time object stored in the df instead of a string. We can also make the date the index of the df which makes getting the rows we want easier.

https://www.dataquest.io/blog/datetime-in-pandas/

I'm not saying that we need to update anything, but it might be worth seeing if parsing the dates when we read the CSV gives us a dataframe that is easier to work with.